lib/remap-functions/src/push.rs

```diff
@@ -0,0 +1,103 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Push;
+
+impl Function for Push {
+    fn identifier(&self) -> &'static str {
+        "push"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Array(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "item",
+                accepts: |_| true,
+                required: true,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+        let item = arguments.required("item")?.boxed();
+
+        Ok(Box::new(PushFn { value, item }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct PushFn {
+    value: Box<dyn Expression>,
+    item: Box<dyn Expression>,
+}
+
+impl Expression for PushFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let mut list = self.value.execute(state, object)?.try_array()?;
+        let item = self.item.execute(state, object)?;
+
+        list.push(item);
+
+        Ok(list.into())
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        use value::Kind;
+
+        self.value
+            .type_def(state)
+            .fallible_unless(Kind::Array)
+            .merge(self.item.type_def(state))
+            .with_constraint(Kind::Array)
+            .with_inner_type(None)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use value::Kind;
+
+    test_type_def![
+        value_array_infallible {
+            expr: |_| PushFn {
+                value: Array::from(vec!["foo", "bar"]).boxed(),
```

Comment:
nit: you used `lit!` and `array!` in `append` before, you can use them here as well.